internal/backend/secrets/svc.go

```diff
@@ -99,6 +125,10 @@ func (s secrets) List(ctx context.Context, scope, key string) ([]string, error)
 	return maps.Keys(data), nil
 }
 
+func limitContext(ctx context.Context, timeout time.Duration) (context.Context, context.CancelFunc) {
```

Comment:
Don't you have a generic fund that already does that?

Reply:
There's something similar in `cmd/ak/common` but it's not exactly the same: this func takes an existing context and limits it, instead of creating a new one. We could think about a common implementation for both in the future, but it's beyond the scope of this PR and not a high priority at this time.